src/state_flow/core.clj

```
@@ -124,3 +126,32 @@
   "Transform a flow step into a state transition function"
   [flow]
   (fn [s] (state/exec flow s)))
+
+(defmacro deftest
+  "Define a test to be run.
+
+  Receives a function `initialize-state`, a function with no
+  arguments that returns the initial state."
+  [sym descr initialize-state cleanup & flows]
```

I think it would be more future proof to require a parameter map with optionals, it is better to extend further this way.

For instance:
- descr might be optional because you are assigning a symbol and the default description can be `(str sym)`.
- Default cleanup might be `identity`
- Default initialize should be `{}`, also the state is expected to be a map, might be nice to assert this

You could (but should you?) do initializing and cleanup inside the flow itself. So it might be good to make it optional.

---

Another thing that it is useful to add: the default flow runner. Currently you can run with `run` or `run!`. Having `run!` as default might be good good to avoid silently ignoring exceptions.

---

yeah, I saw `run!` but didn't think it was a good idea to use `clojure.logging`. I like the idea of having `run` returning the reports and delegating presentation to the runner